nilearn/_utils/validation.py

```diff
@@ -0,0 +1,153 @@
+    mask : filename, niimg, NiftiMasker instance, optional default None)
+        Mask to be used on data. If an instance of masker is passed,
+        then its mask will be used. If no mask is it will be computed
+        automatically by a NiftiMasker.
+
+    target_affine : 3x3 or 4x4 matrix, optional (default None)
+        This parameter is passed to image.resample_img. An important use-case
+        of this parameter is for downsampling the input data to a coarser
+        resolution (to speed of the model fit). Please see the related
+        documentation for details.
+
+    target_shape : 3-tuple of integers, optional (default None)
+        This parameter is passed to image.resample_img. Please see the
+        related documentation for details.
+
+    smoothing_fwhm : float, optional (default None)
+        If smoothing_fwhm is not None, it gives the full-width half maximum in
+        millimeters of the spatial smoothing to apply to the signal.
+
+    standardize : bool, optional (default True):
+        If set, then the data (X, y) are centered to have mean zero along
+        axis 0. This is here because nearly all linear models will want
+        their data to be centered.
+
+    mask_strategy: {'background' or 'epi'}, optional
+        The strategy used to compute the mask: use 'background' if your
+        images present a clear homogeneous background, and 'epi' if they
+        are raw EPI images. Depending on this value, the mask will be
+        computed from masking.compute_background_mask or
+
+    memory: instance of joblib.Memory or string
+        Used to cache the masking process.
+        By default, no caching is done. If a string is given, it is the
+        path to the caching directory.
+
+    memory_level: integer, optional (default 1)
+        Rough estimator of the amount of memory used by caching. Higher value
+        means more memory for caching.
+
+    Returns
+    -------
+    masker : NiftiMasker or MultiNiftiMasker instance
+        Mask to be used on data.
+    """
+    # mask is an image, not a masker
+    if isinstance(mask, _basestring) or (mask is None):
+        masker = NiftiMasker(mask_img=mask,
+                             smoothing_fwhm=smoothing_fwhm,
+                             target_affine=target_affine,
+                             target_shape=target_shape,
+                             standardize=standardize,
+                             mask_strategy=mask_strategy,
+                             memory=memory,
+                             memory_level=memory_level)
+    # mask is a masker object
+    elif isinstance(mask, (NiftiMasker, MultiNiftiMasker)):
+        try:
+            masker = clone(mask)
+            if hasattr(mask, 'mask_img_'):
+                mask_img = mask.mask_img_
+                masker.set_params(mask_img=mask_img)
+                masker.fit()
+        except TypeError as e:
+            # Workaround for a joblib bug: in joblib 0.6, a Memory object
```

Comment:
But then, in this case, `mask.mask_img` is not updated... This should go above line 80 I believe.